Bottom: d995c39f8b1f294fc8149370c6050c6f60b347da
Top:    97f0bb867f14c12f53c1b0ec28cfd8ca09d18f26
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-27 00:09:29 +0100

Refresh of patch

---

diff --git a/src/find.c b/src/find.c
index 5956a2f..bfc0e14 100644
--- a/src/find.c
+++ b/src/find.c
@@ -498,13 +498,13 @@ InitComponentLookup (void)
   /* initialize pad data; start by counting the total number
    * on each of the two possible layers
    */
-  NumberOfPads[COMPONENT_LAYER] = NumberOfPads[SOLDER_LAYER] = 0;
+  NumberOfPads[TOP_SIDE] = NumberOfPads[BOTTOM_SIDE] = 0;
   ALLPAD_LOOP (PCB->Data);
   {
     if (TEST_FLAG (ONSOLDERFLAG, pad))
-      NumberOfPads[SOLDER_LAYER]++;
+      NumberOfPads[BOTTOM_SIDE]++;
     else
-      NumberOfPads[COMPONENT_LAYER]++;
+      NumberOfPads[TOP_SIDE]++;
   }
   ENDALL_LOOP;
   for (i = 0; i < 2; i++)
@@ -634,8 +634,8 @@ LOCtoPVpad_callback (const BoxType * b, void *cl)
 
   if (!TEST_FLAG (i->flag, pad) && IS_PV_ON_PAD (i->pv, pad) &&
       !TEST_FLAG (HOLEFLAG, i->pv) &&
-      ADD_PAD_TO_LIST (TEST_FLAG (ONSOLDERFLAG, pad) ? SOLDER_LAYER :
-                       COMPONENT_LAYER, pad, i->flag))
+      ADD_PAD_TO_LIST (TEST_FLAG (ONSOLDERFLAG, pad) ? BOTTOM_SIDE :
+                       TOP_SIDE, pad, i->flag))
     longjmp (i->env, 1);
   return 0;
 }
@@ -1689,7 +1689,7 @@ LOCtoArcPad_callback (const BoxType * b, void *cl)
   struct lo_info *i = (struct lo_info *) cl;
 
   if (!TEST_FLAG (i->flag, pad) && i->layer ==
-      (TEST_FLAG (ONSOLDERFLAG, pad) ? SOLDER_LAYER : COMPONENT_LAYER)
+      (TEST_FLAG (ONSOLDERFLAG, pad) ? BOTTOM_SIDE : TOP_SIDE)
       && ArcPadIntersect (i->arc, pad) && ADD_PAD_TO_LIST (i->layer, pad, i->flag))
     longjmp (i->env, 1);
   return 0;
@@ -1823,7 +1823,7 @@ LOCtoLinePad_callback (const BoxType * b, void *cl)
   struct lo_info *i = (struct lo_info *) cl;
 
   if (!TEST_FLAG (i->flag, pad) && i->layer ==
-      (TEST_FLAG (ONSOLDERFLAG, pad) ? SOLDER_LAYER : COMPONENT_LAYER)
+      (TEST_FLAG (ONSOLDERFLAG, pad) ? BOTTOM_SIDE : TOP_SIDE)
       && LinePadIntersect (i->line, pad) && ADD_PAD_TO_LIST (i->layer, pad, i->flag))
     longjmp (i->env, 1);
   return 0;
@@ -1958,7 +1958,7 @@ LOCtoPad_callback (const BoxType * b, void *cl)
   struct rat_info *i = (struct rat_info *) cl;
 
   if (!TEST_FLAG (i->flag, pad) && i->layer ==
-	(TEST_FLAG (ONSOLDERFLAG, pad) ? SOLDER_LAYER : COMPONENT_LAYER) &&
+	(TEST_FLAG (ONSOLDERFLAG, pad) ? BOTTOM_SIDE : TOP_SIDE) &&
       ((pad->Point1.X == i->Point->X && pad->Point1.Y == i->Point->Y) ||
        (pad->Point2.X == i->Point->X && pad->Point2.Y == i->Point->Y) ||
        ((pad->Point1.X + pad->Point2.X) / 2 == i->Point->X &&
@@ -2106,7 +2106,7 @@ LOCtoPadPad_callback (const BoxType * b, void *cl)
   struct lo_info *i = (struct lo_info *) cl;
 
   if (!TEST_FLAG (i->flag, pad) && i->layer ==
-      (TEST_FLAG (ONSOLDERFLAG, pad) ? SOLDER_LAYER : COMPONENT_LAYER)
+      (TEST_FLAG (ONSOLDERFLAG, pad) ? BOTTOM_SIDE : TOP_SIDE)
       && PadPadIntersect (pad, i->pad) && ADD_PAD_TO_LIST (i->layer, pad, i->flag))
     longjmp (i->env, 1);
   return 0;
@@ -2225,7 +2225,7 @@ LOCtoPolyPad_callback (const BoxType * b, void *cl)
   struct lo_info *i = (struct lo_info *) cl;
 
   if (!TEST_FLAG (i->flag, pad) && i->layer ==
-      (TEST_FLAG (ONSOLDERFLAG, pad) ? SOLDER_LAYER : COMPONENT_LAYER)
+      (TEST_FLAG (ONSOLDERFLAG, pad) ? BOTTOM_SIDE : TOP_SIDE)
       && IsPadInPolygon (pad, i->polygon))
     {
       if (ADD_PAD_TO_LIST (i->layer, pad, i->flag))
@@ -2680,8 +2680,8 @@ PrintAndSelectUnusedPinsAndPadsOfElement (ElementType *Element, FILE * FP, int f
             if (ADD_PV_TO_LIST (pin, flag))
               return true;
             DoIt (flag, true, true);
-            number = PadList[COMPONENT_LAYER].Number
-              + PadList[SOLDER_LAYER].Number + PVList.Number;
+            number = PadList[TOP_SIDE].Number
+              + PadList[BOTTOM_SIDE].Number + PVList.Number;
             /* the pin has no connection if it's the only
              * list entry; don't count vias
              */
@@ -2721,11 +2721,11 @@ PrintAndSelectUnusedPinsAndPadsOfElement (ElementType *Element, FILE * FP, int f
       {
         int i;
         if (ADD_PAD_TO_LIST (TEST_FLAG (ONSOLDERFLAG, pad)
-                             ? SOLDER_LAYER : COMPONENT_LAYER, pad, flag))
+                             ? BOTTOM_SIDE : TOP_SIDE, pad, flag))
           return true;
         DoIt (flag, true, true);
-        number = PadList[COMPONENT_LAYER].Number
-          + PadList[SOLDER_LAYER].Number + PVList.Number;
+        number = PadList[TOP_SIDE].Number
+          + PadList[BOTTOM_SIDE].Number + PVList.Number;
         /* the pin has no connection if it's the only
          * list entry; don't count vias
          */
@@ -2816,8 +2816,8 @@ PrintElementConnections (ElementType *Element, FILE * FP, int flag, bool AndDraw
     DoIt (flag, true, AndDraw);
     /* printout all found connections */
     PrintPinConnections (FP, true);
-    PrintPadConnections (COMPONENT_LAYER, FP, false);
-    PrintPadConnections (SOLDER_LAYER, FP, false);
+    PrintPadConnections (TOP_SIDE, FP, false);
+    PrintPadConnections (BOTTOM_SIDE, FP, false);
     fputs ("\t}\n", FP);
     if (PrepareNextLoop (FP))
       return (true);
@@ -2835,14 +2835,14 @@ PrintElementConnections (ElementType *Element, FILE * FP, int flag, bool AndDraw
         fputs ("\t\t__CHECKED_BEFORE__\n\t}\n", FP);
         continue;
       }
-    layer = TEST_FLAG (ONSOLDERFLAG, pad) ? SOLDER_LAYER : COMPONENT_LAYER;
+    layer = TEST_FLAG (ONSOLDERFLAG, pad) ? BOTTOM_SIDE : TOP_SIDE;
     if (ADD_PAD_TO_LIST (layer, pad, flag))
       return true;
     DoIt (flag, true, AndDraw);
     /* print all found connections */
     PrintPadConnections (layer, FP, true);
     PrintPadConnections (layer ==
-                         (COMPONENT_LAYER ? SOLDER_LAYER : COMPONENT_LAYER),
+                         (TOP_SIDE ? BOTTOM_SIDE : TOP_SIDE),
                          FP, false);
     PrintPinConnections (FP, false);
     fputs ("\t}\n", FP);
@@ -3035,7 +3035,7 @@ ListStart (int type, void *ptr1, void *ptr2, void *ptr3, int flag)
         PadType *pad = (PadType *) ptr2;
         if (ADD_PAD_TO_LIST
             (TEST_FLAG
-             (ONSOLDERFLAG, pad) ? SOLDER_LAYER : COMPONENT_LAYER, pad, flag))
+             (ONSOLDERFLAG, pad) ? BOTTOM_SIDE : TOP_SIDE, pad, flag))
           return true;
         break;
       }
